### PR TITLE
[google-cloud-components]   - update metadata for output

### PR DESCRIPTION
**Pull request by SinaChavoshi**
Adding support to update metadata for output.

### CODE REVIEW

File: components/google-cloud/google_cloud_components/aiplatform/remote_runner.py [hunk 1]
```diff
@@ -46,15 +46,21 @@ def split_args(kwargs: Dict[str, Any]) -> Tuple[Dict[str, Any], Dict[str, Any]]:
...
 
     # Create the artifact dir if it does not exist
     os.makedirs(os.path.dirname(artifact_path), exist_ok=True)
 
     with open(artifact_path, 'w') as f:
         f.write(text)
 
+    # Create the artifact dir if it does not exist
+    os.makedirs(os.path.dirname(metadata_path), exist_ok=True)
+    runtime_artifact = f"{'uri':{text}}"
```
Comment (sasha-gitg) [inline components/google-cloud/google_cloud_components/aiplatform/remote_runner.py:60]: Based on the implementation here: https://github.com/kubeflow/pipelines/blob/a6a725560c30a769da3c49de739c5e008e761a4f/sdk/python/kfp/components/executor.py#L159

It seems like this format should be:

```Python
{
	"artifacts" : {
  		name: [
  			{
  				"name": name,
    			        "uri": text,  # resource name
    			        "metadata": {}
    		         }
  		]
	}
}
```

@neuromage Please confirm. Also should we retrieve the name in the artifact map and the name of the metadata entry from the executor input?

Comment (neuromage) [reply components/google-cloud/google_cloud_components/aiplatform/remote_runner.py:60]: If you're writing this out from your own container, you can also use the ExecutorOutput proto from the IR directly:
https://github.com/kubeflow/pipelines/blob/075b83af9ca9e37f45907ed3514e2ce8dd637321/api/v2alpha1/pipeline_spec.proto#L832

Comment (SinaChavoshi) [reply components/google-cloud/google_cloud_components/aiplatform/remote_runner.py]: Thank you so much for the feedback, this was very helpful.

**Review by sasha-gitg — APPROVED** [review]

**Pull request merged**